### PR TITLE
docs(MIGRATION): Add the section about renamed classes from RxJS 4.

### PR DESCRIPTION
**Pull request by saneyuki**

### CODE REVIEW

**Comment by saneyuki** [comment]
I think we need this guide for the migration document for RxJS 4 users. They may use `new Rx.CompositeDisposable()` or `Rx.Observer.create()`.

In the migration document, we have already written:

> In RxJS 4, there was the idea of a `CompositeSubscription`.

but there is no mention about `CompositeSubscription` which is same as `CompositeDisposable`. This would confuses a reader.

**Comment by saneyuki** [comment]
@blesh 

How do you think about this pull request?

File: MIGRATION.md [hunk 1]
```diff
@@ -10,6 +10,16 @@ three basic goals:
 Meeting the above goals meant breaking changes to the RxJS API, and a complete rewrite means that we had opportunity
 to change/fix/correct things we've wanted to correct about RxJS in general.
 
+## Key Component Classes Are Recomposed
+
+They are similar to other language implemetations of ReactiveX (e.g. RxJava).
```
Comment (staltz) [inline MIGRATION.md]: `implemetations` => `implementations`

Comment (saneyuki) [inline MIGRATION.md]: oops!

**Comment by saneyuki** [comment]
Addressed.

**Comment by blesh** [comment]
Looks good. Merged with 3dea4e5. Thanks @saneyuki

**Pull request closed without merging**